### PR TITLE
[ios, macos] Rename the iOS and macOS SDKs

### PR DESCRIPTION
**Pull request by jmkiley**
Started the rename process (#10385) for iOS and macOS.

cc @colleenmcginnis

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -528,7 +528,7 @@ limitations under the License.
 
 ===========================================================================
 
-Mapbox GL uses portions of the Mapbox iOS SDK, which was derived from the
+Mapbox GL uses portions of the Mapbox Maps SDK for iOS, which was derived from the
```
Comment (1ec5) [inline LICENSE.md]: This is a reference to the legacy Mapbox iOS SDK, which has not been renamed.

File: platform/ios/CHANGELOG.md [hunk 2]
```diff
@@ -154,7 +154,7 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 ## 3.5.2 - April 7, 2017
 
 * Fixed an issue that caused a crash when the user location annotation was presenting a callout view and the map was moved. ([#8686](https://github.com/mapbox/mapbox-gl-native/pull/8686))
-* This release was built with Xcode 8.3.1, which fixed [a significant bitcode issue](http://www.openradar.me/31302382) introduced in Xcode 8.3 that caused Mapbox iOS SDK 3.5.1 to be 2× larger than 3.5.0.
+* This release was built with Xcode 8.3.1, which fixed [a significant bitcode issue](http://www.openradar.me/31302382) introduced in Xcode 8.3 that caused Mapbox Maps SDK for iOS 3.5.1 to be 2× larger than 3.5.0.
```
Comment (1ec5) [inline platform/ios/CHANGELOG.md]: No need to rename references to previous versions of the SDK that were in fact named “Mapbox iOS SDK”. But while you’re here, can you add a blurb to the top of the changelog about the renaming? (Same for the macOS map SDK changelog.)

File: platform/ios/DEVELOPING.md [hunk 3]
```diff
@@ -70,7 +70,7 @@ To add any Objective-C type, constant, or member to the iOS SDK’s public inter
...
+1. Provide full documentation comments. We use [jazzy](https://github.com/realm/jazzy/) to produce the documentation found in the SDK distribution and [on the Mapbox Maps SDK for iOS website](https://www.mapbox.com/ios-sdk/api/). We also recognize that many developers rely on Xcode’s Quick Help feature. jazzy supports Markdown formatting; however, Quick Help supports only [HeaderDoc](https://developer.apple.com/legacy/library/documentation/DeveloperTools/Conceptual/HeaderDoc/intro/intro.html) syntax and a subset of Doxygen syntax. For hyperlinks, use HTML syntax, which is recognized by both tools.
```
Comment (1ec5) [inline platform/ios/DEVELOPING.md]: “On the Mapbox Maps SDK for iOS website” is a bit unwieldy. “On the website for this SDK” would be slightly more wieldy.

File: platform/ios/docs/pod-README.md [hunk 4]
```diff
@@ -94,6 +94,6 @@ class ViewController: UIViewController {
 }
 ```
 
-Full API documentation is included in this package, within the `documentation` folder. For more details, read “[First steps with the Mapbox iOS SDK](https://www.mapbox.com/help/first-steps-ios-sdk/)” and consult the [online examples](https://www.mapbox.com/ios-sdk/examples/).
+Full API documentation is included in this package, within the `documentation` folder. For more details, read “[First steps with the Mapbox Maps SDK for iOS](https://www.mapbox.com/help/first-steps-ios-sdk/)” and consult the [online examples](https://www.mapbox.com/ios-sdk/examples/).
```
Comment (1ec5) [inline platform/ios/docs/pod-README.md]: Heads-up that the URL to [this guide](https://www.mapbox.com/help/first-steps-ios-sdk/) will likely change as soon as it gets retitled to “First steps with the Mapbox Maps SDK for iOS”.

File: platform/macos/CHANGELOG.md [hunk 5]
```diff
@@ -49,7 +49,7 @@
 
 ## 0.5.1
 
-This version of the Mapbox macOS SDK corresponds to version 3.6.4 of the Mapbox iOS SDK.
+This version of the Mapbox Maps SDK for macOS corresponds to version 3.6.4 of the Mapbox Maps SDK for iOS.
```
Comment (1ec5) [inline platform/macos/CHANGELOG.md]: These and subsequent references are to older versions of the iOS and macOS map SDKs that were called “Mapbox iOS SDK” and “Mapbox macOS SDK” (or “Mapbox OS X SDK”), respectively.

File: platform/ios/src/MGLMapView.mm [hunk 6]
```diff
@@ -1975,7 +1975,7 @@ - (void)trackGestureEvent:(NSString *)gestureID forRecognizer:(UIGestureRecogniz
 
 - (void)showAttribution:(__unused id)sender
 {
-    NSString *title = NSLocalizedStringWithDefaultValue(@"SDK_NAME", nil, nil, @"Mapbox iOS SDK", @"Action sheet title");
+    NSString *title = NSLocalizedStringWithDefaultValue(@"SDK_NAME", nil, nil, @"Mapbox Maps SDK for iOS", @"Action sheet title");
```
Comment (1ec5) [inline platform/ios/src/MGLMapView.mm]: This change should be reflected in at least the base Localizable.strings file, if not all the translations as well.

Comment (jmkiley) [reply platform/ios/src/MGLMapView.mm]: Oops, thanks! I don't know why I didn't commit those.

**Comment by jmkiley** [comment]
Thanks @1ec5! I revised according to your feedback.

**Pull request merged**